pkg/networkserver/config.go

```diff
@@ -19,15 +19,23 @@ import (
 
 	"go.thethings.network/lorawan-stack/pkg/errors"
 	"go.thethings.network/lorawan-stack/pkg/ttnpb"
+	"go.thethings.network/lorawan-stack/pkg/types"
 )
 
 // Config represents the NetworkServer configuration.
 type Config struct {
-	Devices             DeviceRegistry         `name:"-"`
-	DownlinkTasks       DownlinkTaskQueue      `name:"-"`
-	DeduplicationWindow time.Duration          `name:"deduplication-window" description:"Time window during which, duplicate messages are collected for metadata"`
-	CooldownWindow      time.Duration          `name:"cooldown-window" description:"Time window starting right after deduplication window, during which, duplicate messages are discarded"`
-	DownlinkPriorities  DownlinkPriorityConfig `name:"downlink-priorities" description:"Downlink message priorities"`
+	Devices                       DeviceRegistry         `name:"-"`
+	DownlinkTasks                 DownlinkTaskQueue      `name:"-"`
+	DeduplicationWindow           time.Duration          `name:"deduplication-window" description:"Time window during which, duplicate messages are collected for metadata"`
+	CooldownWindow                time.Duration          `name:"cooldown-window" description:"Time window starting right after deduplication window, during which, duplicate messages are discarded"`
+	DownlinkPriorities            DownlinkPriorityConfig `name:"downlink-priorities" description:"Downlink message priorities"`
+	NetID                         types.NetID            `name:"net_id" description:"NetID of this Network Server"`
+	DefaultADRMargin              *float32               `name:"default-adr-margin" description:"The default margin Network Server should add in ADR requests if not configured in device's MAC settings"`
```

Comment:
We could also put these in a `Defaults` struct inside the `Config` struct

Reply:
Originally I used `MACSettings` directly for this, but it did not work well with the CLI.
I don't think we should redefine `MACSettings` here

Reply:
`Defaults` doesn't have to be `MACSettings`. I'm strongly in favor of introducing a struct for this.